crates/libcontainer/src/container/builder.rs

```diff
@@ -253,74 +243,66 @@ impl<'a> ContainerBuilder<'a> {
     ///
     /// ```no_run
     /// # use libcontainer::container::builder::ContainerBuilder;
-    /// # use libcontainer::syscall::syscall::create_syscall;
-    /// # use libcontainer::workload::default::DefaultExecutor;
+    /// # use libcontainer::syscall::syscall::SyscallType;
+    /// # use libcontainer::workload::default::get_executor;
     ///
     /// ContainerBuilder::new(
     ///     "74f1a4cb3801".to_owned(),
-    ///     create_syscall().as_ref(),
+    ///     SyscallType::default(),
     /// )
-    /// .with_executor(vec![Box::<DefaultExecutor>::default()]);
+    /// .with_executor(get_executor());
     /// ```
-    pub fn with_executor(
-        mut self,
-        executors: Vec<Box<dyn Executor>>,
-    ) -> Result<Self, LibcontainerError> {
-        if executors.is_empty() {
-            return Err(LibcontainerError::NoExecutors);
-        };
-        self.executor_manager = ExecutorManager { executors };
+    pub fn with_executor(mut self, executor: Executor) -> Result<Self, LibcontainerError> {
+        self.executor = executor;
```

Comment:
Does this still needs to return result? we are simply setting the executor and returning, so can simply return self instead of a result

Reply:
No. Good catch.

Reply:
b220267